src/components/PurchaseOrder/CloseOrder/CloseOrderModal.js

```diff
@@ -35,104 +34,85 @@ const DEFAULT_REASONS = {
     const reasons = closingReasons.map((reason) => ({
       label: reason.value,
       value: reason.value,
     }));
 
-    if (!isVisible) {
-      return null;
-    }
-
     return (
-      <Fragment>
-        <Button
-          buttonStyle="primary"
-          marginBottom0
-          onClick={this.openModal}
-          style={{ marginRight: '10px' }}
-        >
-          <FormattedMessage id="ui-orders.paneBlock.closeBtn" />
-        </Button>
-        <Modal
-          label={<FormattedMessage id="ui-orders.closeOrderModal.title" values={{ orderId }} />}
-          open={this.state.openModal}
-        >
-          <Row>
-            <Col xs={12}>
-              <Select
-                dataOptions={reasons}
-                label={<FormattedMessage id="ui-orders.closeOrderModal.reason" />}
-                onChange={e => this.setState({ reason: e.target.value })}
-                required
-              >
-                {Object.keys(DEFAULT_REASONS).map((key) => (
-                  <FormattedMessage
-                    id={`ui-orders.closeOrderModal.closingReasons.${key}`}
-                    key={key}
-                  >
-                    {(message) => <option value={DEFAULT_REASONS[key]}>{message}</option>}
-                  </FormattedMessage>
-                ))}
-              </Select>
-              <TextArea
-                label={<FormattedMessage id="ui-orders.closeOrderModal.notes" />}
-                onChange={e => this.setState({ note: e.target.value })}
-              />
-            </Col>
-          </Row>
-          <Row>
-            <Col
-              className={css.buttonsLine}
-              xs={12}
+      <Modal
+        label={<FormattedMessage id="ui-orders.closeOrderModal.title" values={{ orderNumber }} />}
+        open={isOpen}
+      >
+        <Row>
+          <Col xs={12}>
+            <Select
+              autoFocus
+              label={<FormattedMessage id="ui-orders.closeOrderModal.reason" />}
+              dataOptions={reasons}
+              onChange={this.reasonChanged}
+              placeholder=" "
+              defaultValue=""
+            >
+              {Object.keys(DEFAULT_REASONS).map((key) => (
+                <FormattedMessage
+                  id={`ui-orders.closeOrderModal.closingReasons.${key}`}
+                  key={key}
+                >
+                  {(message) => <option value={DEFAULT_REASONS[key]}>{message}</option>}
+                </FormattedMessage>
+              ))}
+            </Select>
+            <TextArea
+              label={<FormattedMessage id="ui-orders.closeOrderModal.notes" />}
+              onChange={this.noteChanged}
+            />
+          </Col>
+        </Row>
+        <Row>
+          <Col
+            className={css.buttonsLine}
+            xs={12}
+          >
+            <Button
+              buttonStyle="primary"
+              disabled={!this.state.reason}
+              onClick={() => closeOrderSubmit(this.state)}
```

Comment:
I don't think it's a good idea to pass the whole state. If it will be extended later some internal state could be leaked without a purpose. Plus it's less readable what data callback is expecting.